src/components/NewWorkspaceModal.js

```diff
@@ -118,8 +118,8 @@ const NewWorkspaceModal = Utils.withDisplayName('NewWorkspaceModal', ({
     Ajax(signal).Groups.list().then(setAllGroups),
     !!cloneWorkspace && Ajax(signal).Workspaces.workspace(namespace, cloneWorkspace.workspace.name).checkBucketLocation(cloneWorkspace.workspace.googleProject, cloneWorkspace.workspace.bucketName)
       .then(({ location }) => {
-        // For current phased regionality release, we only allow US workspace buckets.
-        setBucketLocation(isUSLocation(location) ? location : defaultLocation)
+        // For current phased regionality release, we only allow US or NORTHAMERICA-NORTHEAST1 (Montreal) workspace buckets.
+        setBucketLocation(isUSLocation(location) || location === montrealRegion ? location : defaultLocation)
```

Comment:
Nit: Perhaps use 

```suggestion
        setBucketLocation(isUSLocation(location) || _.eq(montrealRegion, value) ? location : defaultLocation)
```

to be consistent with the [other comparison below](https://github.com/DataBiosphere/terra-ui/pull/2709/files#diff-1fcbff2d49cbfcbbbd2e6fc3f78f453a36ae84030cbddfec06885366f9c00bc8R94).